drivers/npmx/npmx_shell.c

```diff
@@ -1671,6 +1680,154 @@ static int cmd_ldsw_ldo_voltage_get(const struct shell *shell, size_t argc, char
+
+	if (err != 0) {
+		shell_error(shell, "Error: instance index and config value must be integers.");
+		return 0;
+	}
+
+	if (ldsw_indx >= NPMX_PERIPH_LDSW_COUNT) {
+		shell_error(shell, "Error: LDSW instance index is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
+	npmx_ldsw_soft_start_config_t soft_start_config;
+	npmx_error_t err_code = npmx_ldsw_soft_start_config_get(ldsw_instance, &soft_start_config);
+
+	if (!check_error_code(shell, err_code)) {
+		shell_error(shell, "Error: unable to read soft-start config.");
+	}
+
+	char *unit = SHELL_STRING_UNIT_NONE;
+	switch (config_type) {
+	case LDSW_SOFT_START_TYPE_ENABLE:
+		soft_start_config.enable = !!config_val;
+		break;
+
+	case LDSW_SOFT_START_TYPE_CURRENT:
+		npmx_ldsw_soft_start_current_t current =
+			npmx_ldsw_soft_start_current_convert(config_val);
+
+		if (current == NPMX_LDSW_SOFT_START_CURRENT_INVALID) {
+			shell_error(shell, "Error: wrong soft-start value.");
+			return 0;
+		}
+		soft_start_config.current = current;
+		unit = SHELL_STRING_UNIT_MA;
+		break;
+	}
+
+	err_code = npmx_ldsw_soft_start_config_set(ldsw_instance, &soft_start_config);
+
+	shell_print(shell, "Success: %d.", soft_start_config.enable);
+
+	if (check_error_code(shell, err_code)) {
+		shell_print(shell, "Success: %s%s.", argv[2], unit);
+	} else {
+		shell_error(shell, "Error: unable to set soft-start value.");
+	}
+
+	return 0;
+}
+
+static int ldsw_soft_start_config_get(const struct shell *shell, size_t argc, char **argv,
+				      ldsw_soft_start_config_type_t config_type)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return 0;
+	}
+
+	if (argc < 2) {
+		shell_error(shell, "Error: missing LDSW instance index.");
+		return 0;
+	}
+
+	int err = 0;
+	uint8_t ldsw_indx = CLAMP(shell_strtoul(argv[1], 0, &err), 0, UINT8_MAX);
+
+	if (err != 0) {
+		shell_error(shell, "Error: LDSW instance must be an integer.");
+		return 0;
+	}
+
+	if (ldsw_indx >= NPMX_PERIPH_LDSW_COUNT) {
+		shell_error(shell, "Error: LDSW instance index is too high: no such instance.");
+		return 0;
+	}
+
+	npmx_ldsw_t *ldsw_instance = npmx_ldsw_get(npmx_instance, ldsw_indx);
+
+	npmx_error_t err_code;
+	npmx_ldsw_soft_start_config_t soft_start_config;
+
+	err_code = npmx_ldsw_soft_start_config_get(ldsw_instance, &soft_start_config);
+
+	if (check_error_code(shell, err_code)) {
+		char *unit = SHELL_STRING_UNIT_NONE;
+		uint32_t config_val;
+		switch (config_type) {
+		case LDSW_SOFT_START_TYPE_ENABLE:
+			config_val = soft_start_config.enable;
+			break;
+		case LDSW_SOFT_START_TYPE_CURRENT:
+			npmx_ldsw_soft_start_current_convert_to_ma(soft_start_config.current,
+								   &config_val);
+			unit = SHELL_STRING_UNIT_MA;
+			break;
+		}
+		shell_print(shell, "Value: %u%s.", config_val, unit);
```

Comment:
I think this is over-engineered.
Simply printing in `case` would be more readable.

Reply:
This could be standardized for all functions and could reduce the code where there are more than two cases.

Reply:
I agree with @masz-nordic , I do not like this solution either. Using string format for print with an empty string seems like an unnecessary waste of time.